src/subscription_manager/certdirectory.py

```diff
@@ -330,6 +330,21 @@ def list_for_product(self, product_id):
                     entitlements.append(cert)
         return entitlements
 
+    def list_for_product_name(self, product_name, case_sensitive=True):
+        """
+        Returns all entitlement certificates providing access to the given
+        product by name.
+        """
+        entitlements = []
+        for cert in self.list():
+            for cert_product in cert.products:
+                if product_name == cert_product.name:
+                    entitlements.append(cert)
+                elif not case_sensitive and product_name.lower() == cert_product.name.lower():
```

Comment:
`lower()` to compare strings is a bad idea, as it will not properly handle the right equivalents between uppercase and lowercase characters in some cases of non-latin languages. Please use [casefold()](https://docs.python.org/3/library/stdtypes.html#str.casefold) instead.